### PR TITLE
feat: Add login screen

### PR DESCRIPTION
**Pull request by AkshatGKMit**
Add login screen and its api calls

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -26,8 +29,9 @@ const Main = () => {
     <>
       <PlatformDependentStatusBar />
       <SafeAreaView style={[globalStyles.screen]}>
-        <Text>Tuneify</Text>
+        <LoginScreen />
```
Comment (02priyanshi) [inline App.tsx]: why login screen is in App.tsx file?

File: src/screens/index.ts
Comment (02priyanshi) [inline src/screens/index.ts:1]: remove this

File: src/screens/loginScreen/LoginScreen.tsx
Comment (02priyanshi) [inline src/screens/loginScreen/LoginScreen.tsx:1]: change file name to login

File: src/screens/loginScreen/LoginScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import { useContext, useEffect } from 'react';
+import { View, Text, Image, TouchableHighlight, Linking } from 'react-native';
+import Toast from 'react-native-toast-message';
```
Comment (02priyanshi) [inline src/screens/loginScreen/LoginScreen.tsx]: make it custom

File: src/screens/loginScreen/LoginScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    endpoints: { account: accountEndpoints },
+  } = ApiConstants;
+
+  const getUserAuthorization = async () => {
+    const { redirectUrl, authorizationScope, authResponseType } = accountData;
+    const { requestAuthorization: requestAuthorizationEndpoint } = accountEndpoints;
+
+    const state = generateRandomString(16);
+
+    const searchParams: UserAuthorizationParams = {
```
Comment (02priyanshi) [inline src/screens/loginScreen/LoginScreen.tsx]: use this in axios file

File: src/screens/loginScreen/LoginScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    const url = ACCOUNT_BASE_URL + requestAccessTokenEndpoint;
+
+    const body: RequestAccessTokenBody = {
+      code,
+      grant_type: grantType,
+      redirect_uri: redirectUrl,
+    };
+
+    const Authorization = `${basic} ${Buffer.from(CLIENT_ID + ':' + CLIENT_SECRET).toString(
```
Comment (02priyanshi) [inline src/screens/loginScreen/LoginScreen.tsx]: create a method for this

File: src/screens/loginScreen/LoginScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+    if (!response.success) {
+      const { message, code } = response.error;
+      Toast.show({ text1: `Error ${code}`, text2: message, type: 'error' });
+      return;
+    }
+  };
+
+  useEffect(() => {
+    const handleDeepLink = ({ url }: { url: string }) => {
```
Comment (02priyanshi) [inline src/screens/loginScreen/LoginScreen.tsx]: this fun should be outside useEffect

**Review by 02priyanshi — CHANGES_REQUESTED** [review]

File: App.tsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ const Main = () => {
     <>
       <PlatformDependentStatusBar />
       <SafeAreaView style={[globalStyles.screen]}>
```
Comment (02priyanshi) [inline App.tsx:31]: remove styles from array of there is only one style

File: src/network/instanceMethods.ts [hunk 7]
```diff
@@ -19,6 +19,12 @@ async function _post<T, Body, Params = {}>(
   return response.data ?? response;
 }
 
+async function _postAccount<T, Body>(url: string, data: Body): Promise<ApiCallResponse<T>> {
```
Comment (02priyanshi) [inline src/network/instanceMethods.ts:22]: why this fun name is still _postAccount

**Review by 02priyanshi — APPROVED** [review]

File: src/network/instanceMethods.ts (hunk 7 above)
Comment (AkshatGKMit) [reply src/network/instanceMethods.ts:22]: i restriced user to add any more headers

**Pull request merged**